src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -360,7 +367,7 @@ protected override void ApplyProjectChanges(ProjectChanges projectChanges)
                 {
                     try
                     {
-                        _applyChangesProjectFile.Save();
+                        _applyChangesProjectFile.SaveAsync(CancellationToken.None).Wait();
```

Comment:
i'm heavily leaning toward us making everything async, and only having the outermost public *legacy* entrypoints wait.  i don't like all these waits scattered everywhere.

Reply:
(replied to second comment)